File: 3079.find-the-sum-of-encrypted-integers.py
#
# @lc app=leetcode id=3079 lang=python3
#
# [3079] Find the Sum of Encrypted Integers
#

# @lc code=start
from typing import List


class Solution:
    def sumOfEncryptedInt(self, nums: List[int]) -> int:
        result = 0
        for num in nums:
            numSet = set()
            for c in str(num): numSet.add(int(c))
            result += int(str(max(numSet)) * len(str(num)))
        return result
        
# @lc code=end

File: 3079.find-the-sum-of-encrypted-integers.cpp
/*
 * @lc app=leetcode id=3079 lang=cpp
 *
 * [3079] Find the Sum of Encrypted Integers
 */

// @lc code=start
#include <vector>
#include <unordered_set>
#include <string>

class Solution {
public:
    int sumOfEncryptedInt(std::vector<int>& nums) {
        int result = 0;
        for (int num : nums) {
            std::unordered_set<int> numSet;
            std::string numString = std::to_string(num);
            for (char c : numString) {
                numSet.insert(c - '0');
            }
            int maxDigit = 0;
            for (int digit : numSet) {
                maxDigit = std::max(maxDigit, digit);
            }
            std::string maxDigitStr(numString.length(), maxDigit + '0');
            result += std::stoi(maxDigitStr);
        }
        return result;
    }
};

// @lc code=end

